class-plugin-autoupdate-filter.php

```diff
@@ -14,29 +14,104 @@ class Plugin_Autoupdate_Filter {
+			add_action(
+				'admin_notices',
+				function() use ( $error_message ) {
+					echo '<div class="notice notice-error"><p><strong> Plugin Autoupdate Filter:</strong> Unable to get autoupdate settings (' . esc_html( $error_message ) . ').</p></div>';
+				}
+			);
+		}
 
 		// setup plugins and core to autoupdate _unless_ it's during specific day/time
 		add_filter( 'auto_update_plugin', array( $this, 'auto_update_specific_times' ), 10, 2 );
 		add_filter( 'auto_update_core', array( $this, 'auto_update_specific_times' ), 10, 2 );
 
 		// Replace automatic update wording on plugin management page in admin
-		add_filter( 'plugin_auto_update_setting_html', array( $this, 'plugin_autoupdate_filter_custom_setting_html' ), 11, 3 );
+		add_filter( 'plugin_auto_update_setting_html', array( $this, 'custom_setting_html' ), 11, 3 );
 
 		//Append text to upgrade text on plugins page for plugins explicitly set to not autoupdate
-		add_action( 'admin_init', array( $this, 'plugin_autoupdate_filter_change_upgrade_message_for_specific_plugins' ) );
+		add_action( 'admin_init', array( $this, 'upgrade_message_for_specific_plugins' ) );
 
 		// Always send auto-update emails to T51 concierge email address
-		add_filter( 'auto_plugin_theme_update_email', array( $this, 'plugin_autoupdate_filter_custom_update_emails' ), 10, 4 );
-		add_filter( 'auto_core_update_email', array( $this, 'plugin_autoupdate_filter_custom_update_emails' ), 10, 4 );
-		add_filter( 'automatic_updates_debug_email', array( $this, 'plugin_autoupdate_filter_custom_debug_email' ), 10, 3 );
+		add_filter( 'auto_plugin_theme_update_email', array( $this, 'custom_update_emails' ), 10, 4 );
+		add_filter( 'auto_core_update_email', array( $this, 'custom_update_emails' ), 10, 4 );
+		add_filter( 'automatic_updates_debug_email', array( $this, 'custom_debug_email' ), 10, 3 );
 
 		// re-enable core update emails which are disabled in an mu-plugin at the Atomic platform level
 		add_filter( 'automatic_updates_send_debug_email', '__return_true', 11 );
 		add_filter( 'auto_core_update_send_email', '__return_true', 11 );
 		add_filter( 'auto_plugin_update_send_email', '__return_true', 11 );
 		add_filter( 'auto_theme_update_send_email', '__return_true', 11 );
 
+		// "Disable all autoupdates" toggle
+		add_filter( 'auto_update_plugin', array( $this, 'maybe_disable_all_autoupdates' ), PHP_INT_MAX, 2 );
+		add_filter( 'auto_update_core', array( $this, 'maybe_disable_all_autoupdates' ), PHP_INT_MAX, 2 );
+		add_filter( 'auto_update_theme', array( $this, 'maybe_disable_all_autoupdates' ), PHP_INT_MAX, 2 );
+		add_action( 'admin_init', array( $this, 'autoupdate_settings_admin_notice' ) );
+
+	}
+
+	/**
+	 * Load settings from the centralized settings page
+	 */
+	private function get_auto_update_settings(): stdClass|WP_Error {
+
+		$response = wp_remote_get(
```

Comment:
Since we're using a static string, `wp_remote_get` is fine ... but I want to propose that we switch to `wp_safe_remote_get` ([source](https://developer.wordpress.org/reference/functions/wp_safe_remote_get/)) simply because it would be better for the future if anything changes 😄

Reply:
Done!